packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -748,11 +737,22 @@ String _nullsafeKotlinTypeForDartType(TypeDeclaration type) {
 }
 
 /// Returns an expression to cast [variable] to [kotlinType].
-String _cast(String variable,
-    {required String kotlinType, bool safeCast = false}) {
+String _cast(String variable, {required TypeDeclaration type}) {
   // Special-case Any, since no-op casts cause warnings.
-  if (kotlinType == 'Any?' || (safeCast && kotlinType == 'Any')) {
+  final String typeString = _kotlinTypeForDartType(type);
+  if (typeString == 'Any?' || (type.isNullable && typeString == 'Any')) {
```

Comment:
With the way this has been reworked, `typeString` can't contain `?`, so the first half of this condition can be removed.